lib/src/DonutChart.js

```diff
@@ -6,218 +6,156 @@
-            this.setState({
-                selected: item
-            });
-            this.props.onMouseEnter(item);
+import LegendItem from './LegendItem';
+
+export const DonutChartContext = createContext();
+
+const DonutChart = ({
+  className, clickToggle, colors, colorFunction, data, formatValues, height, legend,
+  onClick, onMouseEnter, width, emptyColor, strokeColor, ...restProps
+}) => {
+  const [selected, setSelected] = useState(data[0]);
+  const [toggleSelect, setToggleSelect] = useState(false);
+
+  useEffect(() => {
+    setSelected(data[0]);
+    setToggleSelect(false);
+  }, [data]);
+
+  const total = data.reduce((sum, currItem) => sum + currItem.value, 0);
+  const graphWidth = legend ? (width * (2 / 3)) : width;
+
+  const itemProps = (item, index) => {
+    const { isEmpty, label } = item;
+    const isSelected = selected.label === label;
+    const isToggled = isSelected && toggleSelect;
+
+    return {
+      classNames: `${item.className ? item.className : ''} ${isEmpty ? 'empty' : ''} ${isSelected ? 'selected' : ''} ${isToggled ? 'toggled' : ''}`.trim(),
+      fill: isEmpty ? emptyColor : colorFunction(colors, index),
+      opacity: isSelected && !toggleSelect ? 0.5 : 1,
+      stroke: isEmpty ? emptyColor : strokeColor,
+      clickHandlers: {
+        onClick: () => {
+          if (selected.label === label) {
+            const toggle = clickToggle ? !toggleSelect : false;
+            setSelected(item);
+            setToggleSelect(toggle);
+            onClick(item, toggle);
+          }
+        },
+        onMouseEnter: () => {
+          if (!toggleSelect) {
+            setSelected(item);
+            onMouseEnter(item);
+          }
         }
-    }
-
-    /* React render function */
-    render() {
-        const {
-            startAngle, width, height, formatValues, className, data, legend,
-            emptyColor, strokeColor, colors, colorFunction, innerRadius,
-            outerRadius, emptyOffset, selectedOffset, toggledOffset
-        } = this.props;
-
-        const arcsClassName = `${className}-arcs`;
-        const innerTextClassName = `${className}-innertext`;
-        const legendClassName = `${className}-legend`;
-
-        const checkData = data.length ? data : [{
-            label: '',
-            value: 100,
-            isEmpty: true
-        }];
-
-        const total = calculateTotal(checkData);
-        const twoThirds = 2 / 3;
-        const graphWidth = legend ? (width * twoThirds) : width;
-        const legendWidth = width - graphWidth;
+      }
+    };
+  };
+
+  return (
+    <DonutChartContext.Provider value={{
+      ...restProps,
+      className,
+      data,
+      graphWidth,
+      itemProps,
+      selected,
+      toggleSelect,
+      total,
+      width
+    }}>
+      <svg className={ className } style={{ height, width }} viewBox={`0 0 ${width} ${height}`}>
+        <Arcs />
+        <g className={ `${className}-innertext` }>
```

Comment:
Previously [the separate `InnerText` component](https://github.com/vonbearshark/react-donut-chart/pull/25/files#diff-53e16879585d151ac49bc39c3a2a460fd8783818341fcbe808cd67dc6b8af9ab)